honeybee_radiance_folder/folderutil.py

```diff
@@ -0,0 +1,314 @@
+
+        Args:
+            name: Name for state.
+            default: Path to file to be used for normal representation of the geometry.
+            direct: Path to file to be used for direct studies.
+        """
+        self.name = name
+        self.default = default
+        self.direct = direct
+
+    @classmethod
+    def from_dict(cls, input_dict):
+        """Create a state from an input dictionary.
+        ```
+        {
+          "name": "grass_covered",
+          "default": "ground..summer..000.rad",
+          "direct": "ground..direct..000.rad",
+        }
+        ```
+        """
+        # TODO: add checks for keys
+        name = input_dict['name']
+        default = input_dict['default']
+        direct = input_dict['direct']
+        return cls(name, default, direct)
+
+    def validate(self, folder):
+        """Validate files in this state."""
+        assert os.path.isfile(os.path.join(folder, self.default)), \
+            'Failed to find default file for %s' % self.name
+        assert os.path.isfile(os.path.join(folder, self.direct)), \
+            'Failed to find direct file for %s' % self.name
+
+    def __repr__(self):
+        return 'GeometryState: {}'.format(self.name)
+
+
+class ApertureState(GeometryState):
+    """A state for a dynamic aperture from Radiance files.
+    
+    Attributes:
+        name
+        default
+        direct
+        black
+        tmtx
+        vmtx
+        dmtx
+    """
+
+    def __init__(self, name, default, direct,
+            black=None, tmtx=None, vmtx=None, dmtx=None):
+        """Aperture State.
+
+        Args:
+            name: Name for state.
+            default: Path to file to be used for normal representation of the geometry.
+            direct: Path to file to be used for direct studies.
+            black: Path to file for blacking out the window.
+            tmtx: Path to file for transmittance matrix.
+            vmtx: Path to file for transmittance matrix.
+            dmtx: Path to file for transmittance matrix.
+        """
+        GeometryState.__init__(self, name, default, direct)
+        self.black = black
+        self.tmtx = tmtx
+        self.vmtx = vmtx
+        self.dmtx = dmtx
+
+    @classmethod
+    def from_dict(cls, input_dict):
+        """Create a state from an input dictionary.
+        ```
+        {
+            "name": "clear",
+            "default": "./south_window..default..000.rad",
+            "direct": "./south_window..direct..000.rad",
+            "black": "./south_window..black.rad",
+            "tmtx": "clear.xml",
+            "vmtx": "./south_window..mtx.rad",
+            "dmtx": "./south_window..mtx.rad"
+        }
+        ```
+        """
+        # TODO: add checks for keys
+        name = input_dict['name']
+        default = os.path.normpath(input_dict['default'])
+        direct = os.path.normpath(input_dict['direct'])
+        try:
+            black = input_dict['black']
+        except KeyError:
```

Comment:
I feel like there is a cleaner way of doing this with one try/except and one for loop instead of 4 try/excepts.  It reminds me of plugin users who copy/paste the same component 4 times instead of using a list input.  If this is all part of the TODO that you have in this method, feel free to address it later.

Reply:
This will make more sense in the future as the reaction to each failure will be unique. In some cases I have to use other available values in other cases raise an error. That will be addressed in TODO in the future.